### ISSUE TITLE
Prevent duplicate listing of languages

### ISSUE THREAD

**Issue by phajdan** [hunspell/hunspell#142]
Some myspell dictionaries are listed two times in language selection in e.g. Thunderbird. Cause of this is the use of both '-' and '_' in filenames of dictionairies.
This has been solved for myspell-nl as you can see below

$ dpkg -L myspell-nl
/.
/usr
/usr/share
/usr/share/myspell
/usr/share/myspell/infos
/usr/share/myspell/infos/ooo
/usr/share/myspell/infos/ooo/myspell-nl
/usr/share/myspell/dicts
/usr/share/hunspell
/usr/share/hunspell/nl.aff
/usr/share/hunspell/nl.dic
/usr/share/doc
/usr/share/doc/myspell-nl
/usr/share/doc/myspell-nl/copyright
/usr/share/doc/myspell-nl/changelog.Debian.gz
/usr/share/myspell/dicts/nl.aff
/usr/share/myspell/dicts/nl-NL.aff
/usr/share/myspell/dicts/nl-SR.aff
/usr/share/myspell/dicts/nl-AN.aff
/usr/share/myspell/dicts/nl-BE.aff
/usr/share/myspell/dicts/nl-AW.aff
/usr/share/myspell/dicts/nl.dic
/usr/share/myspell/dicts/nl-NL.dic
/usr/share/myspell/dicts/nl-SR.dic
/usr/share/myspell/dicts/nl-AN.dic
/usr/share/myspell/dicts/nl-BE.dic
/usr/share/myspell/dicts/nl-AW.dic
/usr/share/hunspell/nl_NL.aff
/usr/share/hunspell/nl_SR.aff
/usr/share/hunspell/nl_AN.aff
/usr/share/hunspell/nl_BE.aff
/usr/share/hunspell/nl_AW.aff
/usr/share/hunspell/nl_NL.dic
/usr/share/hunspell/nl_SR.dic
/usr/share/hunspell/nl_AN.dic
/usr/share/hunspell/nl_BE.dic
/usr/share/hunspell/nl_AW.dic

For other dictionaries like en-gb and de-ch, this is not yet the case, see below:

$ dpkg -L myspell-en-gb
/.
/usr
/usr/share
/usr/share/doc
/usr/share/doc/myspell-en-gb
/usr/share/doc/myspell-en-gb/changelog.Debian.gz
/usr/share/doc/myspell-en-gb/copyright
/usr/share/myspell
/usr/share/myspell/infos
/usr/share/myspell/infos/ooo
/usr/share/myspell/infos/ooo/myspell-en-gb
/usr/share/myspell/dicts
/usr/share/hunspell
/usr/share/hunspell/en_GB.aff
/usr/share/hunspell/en_GB.dic
/usr/share/myspell/dicts/en-GB.dic
/usr/share/myspell/dicts/en_GB.aff
/usr/share/myspell/dicts/en_GB.dic
/usr/share/myspell/dicts/en-GB.aff

$ dpkg -L myspell-de-ch
/.
/usr
/usr/share
/usr/share/hunspell
/usr/share/hunspell/de_CH.aff
/usr/share/hunspell/de_CH.dic
/usr/share/myspell
/usr/share/myspell/dicts
/usr/share/myspell/infos
/usr/share/myspell/infos/ooo
/usr/share/myspell/infos/ooo/myspell-de-ch
/usr/share/doc
/usr/share/doc/myspell-de-ch
/usr/share/doc/myspell-de-ch/changelog.gz
/usr/share/doc/myspell-de-ch/RSR.gz
/usr/share/doc/myspell-de-ch/README
/usr/share/doc/myspell-de-ch/Credits
/usr/share/doc/myspell-de-ch/README.swiss
/usr/share/doc/myspell-de-ch/copyright
/usr/share/doc/myspell-de-ch/changelog.Debian.gz
/usr/share/hunspell/de-CH.aff
/usr/share/hunspell/de-CH.dic
/usr/share/hunspell/de-LI.aff
/usr/share/hunspell/de-LI.dic
/usr/share/hunspell/de_LI.aff
/usr/share/hunspell/de_LI.dic
/usr/share/myspell/dicts/de_CH.aff
/usr/share/myspell/dicts/de_CH.dic
/usr/share/myspell/dicts/de_LI.aff
/usr/share/myspell/dicts/de_LI.dic
/usr/share/myspell/dicts/de-CH.aff
/usr/share/myspell/dicts/de-CH.dic
/usr/share/myspell/dicts/de-LI.aff
/usr/share/myspell/dicts/de-LI.dic

This results in the duplicate languages de_CH and German (CH) being offered. Same goes for en_GB and English (GB).

Benefit for fixing this for all dictionaries is that users will be less confused in which language to choose and the list to choose from will not contain duplicates and therefore be shorter.

Original comment by: pander

Original Ticket: [hunspell/bugs/152](https://sourceforge.net/p/hunspell/bugs/152)

**Comment by PanderOpenTaal** [comment]
A related question is the following. Is it allowed to only ship the files:

    /usr/share/hunspell/nl.aff
    /usr/share/hunspell/nl.dic

and not the links:

    /usr/share/hunspell/nl_NL.aff
    /usr/share/hunspell/nl_NL.dic
    /usr/share/hunspell/nl_BE.aff
    /usr/share/hunspell/nl_BE.dic
    /usr/share/hunspell/nl_SR.aff
    /usr/share/hunspell/nl_SR.dic

Reason for this is that (I hope) only "Dutch" will be offered as supported language and not "Dutch, Netherlands", "Dutch, Flanders" and "Dutch, Suriname". All these countries use the same language and the same spelling. In the last ten years it has only been a nuisance to have all these forms for Dutch which all linked to the same dict anyway.

In other words, will nl.aff and nl.dic be enough that users that have software that look for nl_NL, nl_BE or nl_SR will automatically fall back on simply "nl"?
